python/cudf/cudf/_lib/binaryop.pyx

```diff
@@ -175,20 +175,26 @@ def binaryop(lhs, rhs, op, dtype):
     cdef type_id tid = np_to_cudf_types[np.dtype(dtype)]
     cdef data_type c_dtype = data_type(tid)
 
-    if np.isscalar(lhs) or lhs is None:
+    if isinstance(lhs, Scalar) or np.isscalar(lhs) or lhs is None:
```

Comment:
May be it's better to add Scalar to is_scalar list, not sure what would be the impact.

Reply:
Not a blocker for merging this though.